src/test/java/space/obminyashka/items_exchange/controller/UserControllerIntegrationTest.java

```diff
@@ -141,36 +140,28 @@ void addChild_badTotalAmount_shouldReturnBadRequest() throws Exception {
         assertTrue(Objects.requireNonNull(mvcResult.getResolvedException()).getMessage().contains(expectedErrorMessage));
     }
 
-    @Test
-    @WithMockUser(username = "admin")
-    void updateUserPassword_whenOldPasswordWrong_shouldThrowInvalidDtoException() throws Exception {
+    @ParameterizedTest
+    @WithMockUser(username = "user")
+    @MethodSource("listIncorrectDataPassword")
+    void updateUserPassword_whenDataIncorrect_shouldThrowIllegalArgumentException
+            (String password, String confirmPassword, ResultMatcher status, String response)
+            throws Exception {
         when(userService.findByUsernameOrEmail(any())).thenReturn(Optional.of(user));
```

Comment:
The test mock are miss leading: make validation of using that mock, and you'll see the problem

Reply:
removed the excess